tools/pythonpkg/src/pandas/bind.cpp

```diff
@@ -0,0 +1,144 @@
+
+private:
+	py::object getter;
+};
+
+}; // namespace
+
+static LogicalType BindColumn(PandasBindColumn column_p, PandasColumnBindData &bind_data,
+                              const ClientContext &context) {
+	LogicalType column_type;
+	auto &column = column_p.handle;
+
+	auto &config = DBConfig::GetConfig(context);
+	bind_data.numpy_type = ConvertNumpyType(column_p.type);
+	bool column_has_mask = py::hasattr(column.attr("array"), "_mask");
+
+	if (column_has_mask) {
+		// masked object, fetch the internal data and mask array
+		bind_data.mask = make_uniq<RegisteredArray>(column.attr("array").attr("_mask"));
+	}
+
+	if (bind_data.numpy_type == NumpyNullableType::CATEGORY) {
+		// for category types, we create an ENUM type for string or use the converted numpy type for the rest
+		D_ASSERT(py::hasattr(column, "cat"));
+		D_ASSERT(py::hasattr(column.attr("cat"), "categories"));
+		auto categories = py::array(column.attr("cat").attr("categories"));
+		auto categories_pd_type = ConvertNumpyType(categories.attr("dtype"));
+		if (categories_pd_type == NumpyNullableType::OBJECT) {
+			// Let's hope the object type is a string.
+			bind_data.numpy_type = NumpyNullableType::CATEGORY;
+			auto enum_name = string(py::str(column_p.name));
+			vector<string> enum_entries = py::cast<vector<string>>(categories);
+			idx_t size = enum_entries.size();
+			Vector enum_entries_vec(LogicalType::VARCHAR, size);
+			auto enum_entries_ptr = FlatVector::GetData<string_t>(enum_entries_vec);
+			for (idx_t i = 0; i < size; i++) {
+				enum_entries_ptr[i] = StringVector::AddStringOrBlob(enum_entries_vec, enum_entries[i]);
+			}
+			D_ASSERT(py::hasattr(column.attr("cat"), "codes"));
+			column_type = LogicalType::ENUM(enum_name, enum_entries_vec, size);
+			auto numpy_col = py::array(column.attr("cat").attr("codes"));
+			bind_data.internal_categorical_type = string(py::str(numpy_col.attr("dtype")));
+			bind_data.numpy_col = make_uniq<PandasNumpyColumn>(numpy_col);
+		} else {
+			auto numpy_col = py::array(column.attr("to_numpy")());
+			auto numpy_type = numpy_col.attr("dtype");
+			bind_data.numpy_col = make_uniq<PandasNumpyColumn>(numpy_col);
+			// for category types (non-strings), we use the converted numpy type
+			bind_data.numpy_type = ConvertNumpyType(numpy_type);
+			column_type = NumpyToLogicalType(bind_data.numpy_type);
+		}
+	} else if (bind_data.numpy_type == NumpyNullableType::FLOAT_16) {
+		auto pandas_array = column.attr("array");
+		bind_data.numpy_col = make_uniq<PandasNumpyColumn>(py::array(column.attr("to_numpy")("float32")));
+		bind_data.numpy_type = NumpyNullableType::FLOAT_32;
+		column_type = NumpyToLogicalType(bind_data.numpy_type);
+	} else {
+		auto pandas_array = column.attr("array");
+		if (py::hasattr(pandas_array, "_data")) {
+			// This means we can access the numpy array directly
+			bind_data.numpy_col = make_uniq<PandasNumpyColumn>(column.attr("array").attr("_data"));
+		} else if (py::hasattr(pandas_array, "asi8")) {
```

Comment:
I believe this was the only change required to fix the timezone consumption issue